pkg/kubelet/kubelet.go

```diff
@@ -1150,6 +1152,18 @@ func (kl *Kubelet) GetBoundPods() ([]api.BoundPod, error) {
 	return kl.pods, nil
 }
 
+// GetPodFullName provides the first pod that matches namespace and name, or false
+// if no such pod can be found.
+func (kl *Kubelet) GetPodFullName(namespace, name string) (string, bool) {
```

Comment:
In most cases what the caller wants is probably the BoundPod directly, so that's an improvement.  Let me think through whether there's a potential issue there.  This method is hiding the fact that you can get the same pod from two different sources like "api" and "etcd" (which we were not hiding before), so calling this method is an indicator that you don't care about the source, just the pod identification (for kubelet server api calls which don't clarify by uuid).  It's possible most of those APIs are ill defined anyway.